packages/nc-gui/components/smartsheet/toolbar/CalendarRange.vue

```diff
@@ -0,0 +1,246 @@
+<script lang="ts" setup>
+import { UITypes, isSystemColumn } from 'nocodb-sdk'
+import type { SelectProps } from 'ant-design-vue'
+import {
+  ActiveViewInj,
+  IsLockedInj,
+  IsPublicInj,
+  MetaInj,
+  computed,
+  iconMap,
+  inject,
+  ref,
+  useViewColumnsOrThrow,
+  watch,
+} from '#imports'
+
+const meta = inject(MetaInj, ref())
+
+const { $api } = useNuxtApp()
+
+const activeView = inject(ActiveViewInj, ref())
+
+const isLocked = inject(IsLockedInj, ref(false))
+
+const IsPublic = inject(IsPublicInj, ref(false))
+
+const { loadViewColumns } = useViewColumnsOrThrow()
+
+const { loadCalendarMeta, loadCalendarData, loadSidebarData } = useCalendarViewStoreOrThrow()
+
+const calendarRangeDropdown = ref(false)
+
+watch(
+  () => activeView.value?.id,
+  async (newVal, oldVal) => {
+    if (newVal !== oldVal && meta.value) {
+      await loadViewColumns()
+    }
+  },
+  { immediate: true },
+)
+
+const calendarRange = computed<{ fk_from_column_id: string; fk_to_column_id: string | null }[]>(() => {
```

Comment:
`{ fk_from_column_id: string; fk_to_column_id: string | null }` duplicates several times. define once and use multiple times